andela/krypton-ah-backend
# #162414150 Configure Authors Haven to use PostgreSQL as DB Engine

#### Description of Task to be completed?
Authors Haven comes with support for MongoDB, a NoSQL Database. 
We need to configure a fully featured and optimized SQL database engine for production-ready environments.
Hence, we are using PostgreSQL which is also Open Source.

#### How should this be manually tested?
1. Create a .env file in the root directory and fill in the details of .env-sample file
2. Create a database on your system with the following details
`{
    "username": "postgres",
    "password": "root",
    "database": "authorshaven",
}`
For the test database,
`{
    "username": "postgres",
    "password": "root",
    "database": "authorshaven_test",
}`
3. On your command line, type the command `psql` to initialize the postgres. 
   When the postgres interface is displayed on the command line, type `\l`.
   This shows a list of the databases and `authorshaven` should be one of them with the owner as `postgres`

#### Any background context you want to provide?

This setup will only work if the user has postgres installed already
[Follow this link to set up postgres](https://postgresapp.com/ "Download Postgres")

Graphical clients also helps us display our database information in a well structured format.
[Follow this link to download pgAdmin 4](https://www.pgadmin.org/ "Download PgAdmin")

#### What are the relevant pivotal tracker stories?
[#162414150](https://www.pivotaltracker.com/story/show/162414150)

#### Screenshots (if appropriate)
None

#### Questions
None
